### ISSUE TITLE
[pigeon] Consider removing Obj-C generator

### ISSUE THREAD

**Issue by stuartmorgan** [flutter/flutter#158288]
(Filing to track and centralize discussion, since this has come up in several discussions.)

The number of generators in Pigeon is an issue for ongoing development (e.g., event channel support), and we are currently maintaining two generators—Objective-C and Swift—for the use case of iOS/macOS plugin development. Ideally we would only maintain one, which given the direction of iOS and macOS development would be Swift.

In favor of turning it down:
- We're actively working on [deprecating explicit support for Obj-C in Flutter tooling](https://github.com/flutter/flutter/issues/148586).
- We [want to be using Swift for our own Plugins](https://github.com/flutter/flutter/issues/119015), at which point we would not have any actual first-party usage.

Impediments to turning it down:
- Swift/Obj-C interop is non-trivial; in particular, it doesn't seem to be possible to implement a Swift interface in Obj-C, so at least the top level code (the plugin class, or code factored out of it in cases where that's the whole implementation) would need to be converted to Swift as part of switching to the Swift Pigeon generator. That's work we want to do eventually, but isn't currently a priority.
  - This would apply to third parties as well; very crude GitHub search (e.g., no fork de-duping) suggests a majority of iOS Pigeon clients are using Obj-C rather than Swift, but not an overwhelming majority. Of course, they could always keep using older versions of Pigeon until they had time to transition.
  - Clients would be forced to write/maintain non-generated Swift code, which is not ideal for plugin maintainers who want to use Obj-C based on their own experience.

**Comment by stuartmorgan** [comment]
(Cross ref: Java version is https://github.com/flutter/flutter/issues/158287)

**Comment by ryanheise** [comment]
> We're actively working on [deprecating explicit support for Obj-C in Flutter tooling](https://github.com/flutter/flutter/issues/148586).

My understanding was that this was only for the app project template, not for plugins. https://github.com/ryanheise/just_audio/issues/1368#issuecomment-2619365460

**Comment by stuartmorgan** [comment]
No, all Obj-C templates are deprecated. If you `flutter create -t plugin -i objc` in a current version, you will see a deprecation message.
